soda_machine.rb

```diff
@@ -7,12 +7,22 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    #@sodas.find {|name| name = soda_brand}
+    #@sodas.select {|name| name = soda_brand}
+    #@sodas.detect{|name| name = soda_brand}
+    #@sodas.index(soda_brand)
+    #these are all the methods I attempted to use, but none passed the tests.
```

Comment:
If I looked at the `@sodas` array in irb/pry what would it look like?  What kind of object does the array contain?  

These attempts all assume that it would be an array of strings, like `["Pepsi", "Coke Zero"]`, but it's actually an array of `Soda` objects, so looking for the element that matches your string is always going to fail: `"Pepsi"` will not equal `#<Soda:0x007fd873596020 @brand="Pepsi", @price=1.3>`.

Also, you are using assignment (single `=`) rather than comparisons (`==`).